### ISSUE TITLE
Keyboard intermittently doesn't detect keypresses

### ISSUE THREAD

**Issue by JFMFT** [slayher/CM7_Mecha_issue#18]
This issue has been recurring for at least the last 3 builds of CM7 for the Thunderbolt. A description of the problem in detail:

*As if totally by random, when typing using the stock Android 2.3 Gingerbread keyboard (or any keyboard that is derived from the Android 2.3 keyboard, such as Steven Lin's (author's name) Gingerbread Keyboard, the keyboard will fail to register keypresses. It doesn't ALWAYS do this - just randomly, and intermittently. Sometimes the keyboard works flawlessly, but this problem happens often enough to make it so that I avoid using my phone for any text input where possible (SMS, emails, etc.) because the issue is so pervasive.

*The issue is resolved by turning the screen off and then back on, or SOMETIMES (but not always) by either switching hands, orientations (from portrait to landscape) or setting the phone on a surface for a second, then picking it back up. Usually, as I stated, turning the screen on/off is required, but does not always fix the issue (as in the problem persists even after turning the screen on/off).

*Things I have tried to do to fix the problem, to no avail: 
-wipe the phone completely, reinstalling Slayher's recommended GAPPS and a fresh install of 1.6.1
-clear data/cache (factory reset)
-clear dalvik cache
-use different kernels (imoseyon's, etc.)
-used a different phone (I have two Thunderbolts - and BOTH have this issue!)
-remove ALL applications I could think of that would possibly cause a conflict (including but not limited to ALL secondary keyboard apps, language/dictionary packs, etc.)
-use the phone with a capacitive stylus (yes, it even does this when typing slowly with a capacitive Griffin brand stylus)

**Comment by tmurj33** [comment]
I have this problem too. I have tried everything one thing I have noticed is that its never in the center of the keyboard always (for me anyway) on one of the outside keys

**Comment by skinien** [comment]
This isn't a CM issue.  It happens on Sense ROMs also.  Probably a hardware (digitizer) defect.

**Comment by JFMFT** [comment]
I must simply disagree with you skinien; I have two Thunderbolts with CM7 and experience the issue on both of them. Additionally, I ran one of my Thunderbolts for the last week on Sense and had NO keyboard weirdness/issues.

I'm open to additional dialogue/exploration of this problem if anyone has anything else to contribute/add to this discussion.

**Comment by tmurj33** [comment]
Agreed, I have had two other Thunderbolts with sense and then rooted with sense roms and never had this issue

**Comment by skinien** [comment]
Are you two using the Sense keyboard?  I was using the GB keyboard in the market when I was running Sense ROMs.  It could be the GB keyboard, itself.

Edit:  Sorry, I didn't reread the OP in a while.  It looks like the author agrees that it's an Android 2.3 keyboard issue.  If it's an Android bug, are we calling it a CM bug as well?

**Comment by tmurj33** [comment]
Im calling a CM issue, just because I have never had this issue with sense roms on now 3 phones

**Comment by uhbill** [comment]
I have experienced this on numerous roms, always while using the GB keyboard. So, yes, I believe the issue lies with the keyboard, but that it should thus be correctable within CM.
